Wikipedia/AppDelegate.m

```diff
@@ -60,11 +63,21 @@ - (WMFLegacyImageDataMigration*)imageMigration {
     return _imageMigration;
 }
 
+- (SavedArticlesFetcher*)savedArticlesFetcher {
```

Comment:
I don't think this is a good pattern to start. A while ago we pulled a bunch of logic out of the app delegate and scoped the app delegates purpose to simply:

- Being the delegate of the app (implementing the callbacks from the UIApplication)
- Starting the App View Controller
- Starting Crash reporting/analytics - because we need to start it as soon as possible to catch crashes

Importantly the App Delegate is pretty devoid of business logic (aside from image migration, which actually needs to be moved to where the rest of the migration logic is - in the AppViewController)

Currently, the AppViewController is the place where the data store is inited and other list and model controllers are injected into other view controllers that need them. this basically in the App View Controller's job description - it is responsible for the App up and running by injecting the model controllers and models into the view controllers. 

Having said that, if you want to propose a new architecture for holding onto long living model controllers / business logic-y classes in a place different from the App View Controller, I'd be open to it and we should definitely discus at code review. In the mean time we should keep such logic in the same place and not start competing architectures.